operators/pkg/instctrl/cloudinit.go

```diff
@@ -61,12 +54,20 @@ func (r *InstanceReconciler) EnforceCloudInitSecret(ctx context.Context) error {
 	}
 	log.V(utils.LogDebugLevel).Info("public keys correctly retrieved")
 
-	userdata, err := forge.CloudInitUserData(r.ServiceUrls.NextcloudBaseURL, user, password, publicKeys)
+	server, path, err := r.GetNFSSpecs(ctx)
+
+	if err != nil {
+		if env.MountPathEnabled {
```

Comment:
It would be better to avoid retrieving the NFSSpecs altogether if mounting is not enabled.

Reply:
Fixed